lib/parse.js

```diff
@@ -148,7 +152,7 @@ var parseObject = function (chain, val, options, valuesParsed) {
 };
 
 var parseKeys = function parseQueryStringKeys(givenKey, val, options, valuesParsed) {
-    if (!givenKey) {
+    if (options.allowEmptyKeys ? givenKey !== '' && !givenKey : !givenKey) {
```

Comment:
```suggestion
    if (!givenKey && (!options.allowEmptyKeys || givenKey !== '')) {
```
to avoid the inline ternary